src/MeasurementCodeInjector.php

```diff
@@ -37,11 +55,90 @@ public function setActivePlugins() {
+
+            /**
+             * Adds event listener to element that corresponds to the event
+             *
+             * @param config - configuration for a trackable event
+             */
+            function addListeners(config) {
+                let nodeList = document.querySelectorAll(config.selector);
+                for(node of nodeList) {
+                    if(!ninjaFormsAddedNodes.includes(node)) {
+                        node.addEventListener(config.on, function () {
+                            alert('Got an event called: '.concat(config.action));
+                        });
+                        if(config.pluginName == 'Ninja Forms') ninjaFormsAddedNodes.push(node);
+                    }
+                }
+            }
+
+            /**
+             * Adds first or second layer event listeners on DOM loaded
+             */
+            jQuery(function($){
+                let eventConfigurations = <?php echo json_encode($this->eventConfigurations); ?>;
+                for(config of eventConfigurations) {
+                    if(config.secondLayerOn === null) {
+                        addListeners(config);
+                    }else{
+                        let secondLayerNode;
+                        switch(config.secondLayerSelector) {
+                            case 'document':
+                                secondLayerNode = $(document);
+                                break;
+                            case 'document.body':
+                                secondLayerNode = $(document.body);
+                                break;
+                            default:
+                                secondLayerNode = $(config.secondLayerSelector);
+                                break;
+                        }
+                        const configCopy = JSON.parse(JSON.stringify(config));
+                        secondLayerNode.on(config.secondLayerOn, function(){
+                            addListeners(configCopy);
+                        });
+                    }
+                }
+            });
+        </script>
+        <?php
+    }
+
+    /**
+     * Creates a config object from MeasurementEvent object and adds it to list to be injected
+     *
+     * @param $measurementEvent - MeasurementEvent
+     */
+    private function configureEvent($measurementEvent) {
+        $newEvent = array();
+        $newEvent['pluginName'] = $measurementEvent->getPluginName();
+        $newEvent['category'] = $measurementEvent->getCategory();
+        $newEvent['action'] = $measurementEvent->getAction();
+        $newEvent['selector'] = $measurementEvent->getSelector();
+        $newEvent['on'] = $measurementEvent->getOnEvent();
+        $newEvent['secondLayerSelector'] = $measurementEvent->getSecondLayerSelector();
+        $newEvent['secondLayerOn'] = $measurementEvent->getSecondLayerOnEvent();
+        array_push($this->eventConfigurations, $newEvent);
```

Comment:
This conversion might not be needed if we directly convert the object to a JSON
https://www.codebyamir.com/blog/object-to-json-in-php